plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2740,7 +2740,8 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
 
         RowDelta rowDelta = transaction.newRowDelta();
         table.getSnapshotId().map(icebergTable::snapshot).ifPresent(s -> rowDelta.validateFromSnapshot(s.snapshotId()));
-        TupleDomain<IcebergColumnHandle> dataColumnPredicate = table.getEnforcedPredicate().filter((column, domain) -> !isMetadataColumnId(column.getId()));
+        TupleDomain<IcebergColumnHandle> dataColumnPredicate = table.getEnforcedPredicate().intersect(table.getUnenforcedPredicate())
```

Comment:
Why do we intersect enforced with the unenforced predicate?
They are rather unrelated.

Reply:
My understanding is that both carriers information about predicate.
The difference is that taken for @raunaqmorarka explanation:
```
enforced is the part of the predicate which is guaranteed to be satisfied by the connector, so the engine will not apply it on it's side.
unenforced is the part of the predicate which connector cannot guarantee even if it is able to use it to reduce output, so the engine will apply it on the connector output
```
@raunaqmorarka Correct me if I'm wrong here

Reply:
```
{2:part:date=[ SortedRangeSet[type=date, ranges=1, {[2024-01-01]}] ]}
```

I'm checking `io.trino.plugin.iceberg.TestIcebergLocalConcurrentWritesTest#testConcurrentUpdateWithOverlappingPartitionTransformation` (BTW really cool that we have this battery of new concurrency tests) 

It's unclear to me why the partition predicate is not an "enforced predicate" while debugging your code.

Reply:
Adding such predicates for transformed columns as enforced, would mean pushdowns of those values and connector would need to filter files and rows also during reading from such table as well. It's not supported right now and would require more work.

Reply:
>They are rather unrelated.

We already join those in `IcebergSplitSource`
https://github.com/trinodb/trino/blob/61a79dd875fa48e2e0c2b3230051a526ecd51ad7/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java#L259

Reply:
We are able to prune files in IcebergSplitSource based on unenforced filter, but not rows (hence "unenforced"). Filters on identity partitioned columns are enforced at the row level due to the data being partitioned by value and `PartitionConstraintMatcher` in IcebergSplitSource.
I think it's valid to intersect unenforced filter here because `org.apache.iceberg.MergingSnapshotProducer` is internally checking whether any files matching given filter have been added to the table.